google-ads-examples/src/main/java/com/google/ads/googleads/examples/hotelads/AddHotelCallout.java

```diff
@@ -0,0 +1,231 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.ads.googleads.examples.hotelads;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import com.beust.jcommander.Parameter;
+import com.google.ads.googleads.examples.utils.ArgumentNames;
+import com.google.ads.googleads.examples.utils.CodeSampleParams;
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v2.common.HotelCalloutFeedItem;
+import com.google.ads.googleads.v2.enums.ExtensionTypeEnum.ExtensionType;
+import com.google.ads.googleads.v2.errors.GoogleAdsError;
+import com.google.ads.googleads.v2.errors.GoogleAdsException;
+import com.google.ads.googleads.v2.resources.CampaignExtensionSetting;
+import com.google.ads.googleads.v2.resources.ExtensionFeedItem;
+import com.google.ads.googleads.v2.services.CampaignExtensionSettingOperation;
+import com.google.ads.googleads.v2.services.CampaignExtensionSettingServiceClient;
+import com.google.ads.googleads.v2.services.ExtensionFeedItemOperation;
+import com.google.ads.googleads.v2.services.ExtensionFeedItemServiceClient;
+import com.google.ads.googleads.v2.services.MutateCampaignExtensionSettingsResponse;
+import com.google.ads.googleads.v2.services.MutateExtensionFeedItemsResponse;
+import com.google.ads.googleads.v2.utils.ResourceNames;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.StringValue;
+
+/**
+ * Adds a hotel callout to a specific campaign
+ */
+
+ public class AddHotelCallout {
+
+    private static class AddHotelCalloutParams extends CodeSampleParams {
+
+        public static final String CALLOUT_TEXT = "--calloutText";
+        public static final String LANGUAGE_CODE = "--languageCode";
+
+        @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+        private Long customerId;
+
+        @Parameter(names = ArgumentNames.CAMPAIGN_ID, required = true)
+        private Long campaignId;
+
+        @Parameter(names = CALLOUT_TEXT, required = true)
+        private String calloutText;
+
+        // IETF BCP 47 compliant language code
```

Comment:
Can we link users to a list of the accepted language codes?

Reply:
+1

Reply:
So far all I can find is https://tools.ietf.org/html/bcp47, which is the actual spec, not just a list of codes.  Maybe https://www.w3.org/International/articles/language-tags/ is better?  Should the acceptable codes just be the same as what is returned by https://developers.google.com/google-ads/api/fields/v2/language_constant?authuser=1#language_constant.code ?

Reply:
Can you check with the Eng team to see if it's true?
If the language_constant.code  is what we should use, please include that info here instead of IETF BCP 47 compliant code, which is more indirect.

Reply:
Talked to eng and was pointed to https://developers.google.com/hotels/hotel-ads/api-reference/language-codes